File: rabbitMQ/ruby/send.rb
require 'bunny'

connection = Bunny.new(automatically_recover: false, hostname: 'rabbitmq')
connection.start

channel = connection.create_channel
queue = channel.queue('hello')

begin
  loop do
    puts "Message: "
    message = gets.chomp
    channel.default_exchange.publish(message, routing_key: queue.name)
  end
rescue Interrupt => _
  puts "\nExiting..."
ensure
  connection.close
end

File: rabbitMQ/python/send.py
import pika

# Establish connection to RabbitMQ (outside the loop for efficiency)
connection = pika.BlockingConnection(
    pika.ConnectionParameters(host='rabbitmq'))
channel = connection.channel()

# Declare the queue (if it doesn't exist)
channel.queue_declare(queue='hello')

print("Press Ctrl+C to quit")
try:
    while True:
        message = input("Message: ")
        channel.basic_publish(exchange='', routing_key='hello', body=message)

except KeyboardInterrupt:
    print("\nExiting...")

finally:
    # Close the connection gracefully
    connection.close()

